$DragonFly: src/secure/usr.sbin/sshd/Attic/loginrec.c.patch,v 1.1 2004/07/31 20:05:00 geekgod Exp $
--- loginrec.c.orig	2004-07-23 14:26:51.000000000 +0200
+++ loginrec.c	2004-07-23 14:20:00.000000000 +0200
@@ -652,7 +652,8 @@
 	/* Use strncpy because we don't necessarily want null termination */
 	strncpy(ut->ut_name, li->username, MIN_SIZEOF(ut->ut_name, li->username));
 # ifdef HAVE_HOST_IN_UTMP
-	strncpy(ut->ut_host, li->hostname, MIN_SIZEOF(ut->ut_host, li->hostname));
+	realhostname_sa(ut->ut_host, sizeof ut->ut_host,
+	    &li->hostaddr.sa, li->hostaddr.sa.sa_len);
 # endif
 # ifdef HAVE_ADDR_IN_UTMP
 	/* this is just a 32-bit IP address */
